src/test/java/SearchServiceFilteringTest.java

```diff
@@ -9,57 +9,57 @@
-    }
-    // [END search_client]
+  // [START search_client]
+  private static SearchServiceClient getSearchServiceClient() throws IOException {
+    SearchServiceSettings settings = SearchServiceSettings.newBuilder()
+        .setEndpoint(ENDPOINT)
+        .build();
+    return SearchServiceClient.create(settings);
+  }
+  // [END search_client]
 
-    private static final String DEFAULT_BRANCH_NAME =
-            DEFAULT_CATALOG_NAME + "/branches/default_branch";
+  private static final String DEFAULT_BRANCH_NAME =
+      DEFAULT_CATALOG_NAME + "/branches/default_branch";
 
-    // [START search_product_filter]
-    public static SearchResponse searchFilteredProducts(String query, int pageSize,
-                                                        String filter) throws IOException, InterruptedException {
-        SearchServiceClient searchClient = getSearchServiceClient();
+  // [START search_product_filter]
+  public static SearchResponse searchFilteredProducts(String query, int pageSize,
+      String filter) throws IOException, InterruptedException {
+    SearchServiceClient searchClient = getSearchServiceClient();
 
-        SearchRequest searchRequest = SearchRequest.newBuilder()
-                .setPlacement(DEFAULT_SEARCH_PLACEMENT_NAME)
-                .setBranch(DEFAULT_BRANCH_NAME)
-                .setVisitorId(VISITOR_ID)
-                .setQuery(query)
-                .setPageSize(pageSize)
-                .setFilter(filter)
-                .build();
-        System.out.println("Search with filtering, request: " + searchRequest);
-        SearchResponse response = searchClient.search(searchRequest).getPage().getResponse();
+    SearchRequest searchRequest = SearchRequest.newBuilder()
+        .setPlacement(DEFAULT_SEARCH_PLACEMENT_NAME)
+        .setBranch(DEFAULT_BRANCH_NAME)
+        .setVisitorId(VISITOR_ID)
+        .setQuery(query)
+        .setPageSize(pageSize)
+        .setFilter(filter)
+        .build();
+    System.out.println("Search with filtering, request: " + searchRequest);
+    SearchResponse response = searchClient.search(searchRequest).getPage().getResponse();
 
-        searchClient.shutdownNow();
-        searchClient.awaitTermination(2, TimeUnit.SECONDS);
-        System.out.println("Search with filtering, response: " + response);
-        return response;
-    }
-    // [END search_product_filter]
+    searchClient.shutdownNow();
+    searchClient.awaitTermination(2, TimeUnit.SECONDS);
+    System.out.println("Search with filtering, response: " + response);
+    return response;
+  }
+  // [END search_product_filter]
 
-    @Test
-    public void search() throws IOException, InterruptedException {
+  @Test
+  public void search() throws IOException, InterruptedException {
 
-        searchFilteredProducts("Nest_Maxi", 10,
-                               "(colorFamily: ANY(\"grey\")) AND (price>=100)"
-        );
-    }
+    searchFilteredProducts("Nest_Maxi", 10,
```

Comment:
change the values according to the python samples

Reply:
I don't see an appropriate sample for this file.
It's kind https://github.com/t-karasova/grs-samples-python/blob/master/search/search_with_filtering.py
But not, the file above according https://github.com/t-karasova/grs-retail/blob/update_samples/src/test/java/SearchServiceFacetsExcludedFilterKeyTest.java

For this file, I did refactoring and insert comments.

Reply:
this is SearchServiceFilteringTest file and the value of the  filter should be "'(colorFamily: ANY("Black"))'"

I agree that you left the Facet classes with the existing data as there is no such files in python samples yet

Reply:
Ok, fixed filter variable.